epoxy-integrationtest/src/test/java/com/airbnb/epoxy/EpoxyVisibilityTrackerModelGroupTest.kt

```diff
@@ -0,0 +1,781 @@
+package com.airbnb.epoxy
+
+import android.os.Looper
+import android.view.ViewGroup
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.ext.junit.rules.activityScenarioRule
+import androidx.test.runner.AndroidJUnit4
+import com.airbnb.epoxy.integrationtest.R
+import com.airbnb.epoxy.integrationtest.TestActivity
+import com.airbnb.epoxy.models.TrackerTestModel
+import com.airbnb.epoxy.models.trackerTestModel
+import com.airbnb.epoxy.models.trackerTestModelGroup
+import com.airbnb.epoxy.utils.VisibilityAssertHelper
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+import kotlin.math.roundToInt
+
+/**
+ * Tests visibility tracking of [EpoxyModelGroup]s inside of a [RecyclerView].
+ */
+@Config(sdk = [21], qualifiers = "h831dp-mdpi")
+@RunWith(AndroidJUnit4::class)
+class EpoxyVisibilityTrackerModelGroupTest {
+
+    @get:Rule
+    var activityRule = activityScenarioRule<TestActivity>()
```

Comment:
So I am not super familiar with `activityRule`, do you think we should migrate the existing tests in `epoxy-adapter` to `epoxy-integrationtest` too? They only rely on Robolectric. I am worried this change will introduce code duplication for the assert helper part.

Reply:
Yeah this module still uses Robolectric. I do think we should move the existing tests over but was planning on doing that in a separate PR to keep this one focused. `ActivityScenario` will be the replacement for the now deprecated `Robolectric.setupActivity()`